File: examples/java/kora-java-scheduling-jdk/src/test/java/ru/tinkoff/kora/example/scheduling/jdk/FixDelaySchedulerTests.java
package ru.tinkoff.kora.example.scheduling.jdk;

import java.time.Duration;
import org.awaitility.Awaitility;
import org.junit.jupiter.api.Test;
import ru.tinkoff.kora.scheduling.jdk.FixedDelayJob;
import ru.tinkoff.kora.test.extension.junit5.KoraAppTest;
import ru.tinkoff.kora.test.extension.junit5.TestComponent;

@KoraAppTest(value = Application.class, components = FixedDelayJob.class)
class FixDelaySchedulerTests {

    @TestComponent
    private FixDelayScheduler scheduler;

    @Test
    void scheduled() {
        Awaitility.await().atMost(Duration.ofSeconds(3)).until(() -> scheduler.getState() > 3);
    }
}

File: examples/kotlin/kora-kotlin-scheduling-jdk/src/test/kotlin/ru/tinkoff/kora/kotlin/example/scheduling/jdk/FixDelaySchedulerTests.kt
package ru.tinkoff.kora.kotlin.example.scheduling.jdk

import org.awaitility.Awaitility
import org.junit.jupiter.api.Test
import ru.tinkoff.kora.scheduling.jdk.FixedDelayJob
import ru.tinkoff.kora.test.extension.junit5.KoraAppTest
import ru.tinkoff.kora.test.extension.junit5.TestComponent
import java.time.Duration

@KoraAppTest(value = Application::class, components = [FixedDelayJob::class])
class FixDelaySchedulerTests {

    @TestComponent
    lateinit var scheduler: FixDelayScheduler

    @Test
    fun scheduled() {
        Awaitility.await().atMost(Duration.ofSeconds(3)).until { scheduler.state > 3 }
    }
}
